### ISSUE TITLE
How to do paging with scan-parallel?

### ISSUE THREAD

**Issue by ulsa** [taoensso/faraday#17]
I'm trying to do paging with `scan-parallel` using `:limit`, but I'm not sure how to specify `:last-prim-kvs` in subsequent calls. Each segment needs its own last key, I presume.

Am I missing something, or is paging not implemented for parallel scan?

**Comment by barkanido** [comment]
Is this something worth fixing for a Faraday noob? Are you open for reviewing PR on this? Any thoughts about a reasonable solution direction?

**Comment by kipz** [comment]
I think it would be nice to get this fixed @barkanido given we have the beginning of an implementation, so I expect PR's would be welcomed by the community.

Having said that however, you could probably just manage the threads and paging yourself and just call `scan` directly, and personally, that's the approach I'd prefer here. 

I have an implementation of a `lazy-paged-query` that will manage query paging automatically, and I'm glad that it was easy to build on top of Faraday, but don't feel that it should be part of the API. Handling thread-pools and paging of large result-sets of scans fits into the same category for me. Of course I don't speak for the community here at all though, it's just my opinion!

**Comment by barkanido** [comment]
@kipz fair enough. Maybe your example deserve a place here as an example people can refer to. Or even in the README. Just a thought. Anyway I was just looking for a way to contribute and found this issue. Maybe a task from the TODO is of higher priority?

**Comment by kipz** [comment]
@joelittlejohn what are your thoughts on all this?

**Comment by joelittlejohn** [comment]
@barkanido Re your question about whether this ticket is a good one for a Faraday noob to tackle, it's probably not :slightly_smiling_face: The existing paging implementation is one of the most complex parts of Faraday and as @kipz mentions people have often found that they prefer to avoid the paging feature altogether and implement their own solution (over which they have more control) outside Faraday.

Is this a feature you need or were you just interested in making a useful contribution? I think the most useful thing to be done for Faraday is better documentation. Better docstrings and/or I think it would be _very_ useful to have a list of examples that show real-world usage covering all typical ways to use Faraday's functions.

**Comment by joelittlejohn** [comment]
For a Faraday noob that wants to contribute something useful, I recommend using the library for a while on a few projects and over time you will inevitably uncover something you'd like but is missing.
